File: Matrizes/ExercicioH.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Algoritmos.Matrizes
{
    // Criar uma matriz A de 20 elementos, ler 20 elementos e armazenar em A.
    // Criar uma matriz B de mesma dimensão de A, onde o primeiro elemento de B é o último de A 
    // e assim por diante.
    internal class ExercicioH
    {
        static void Main(string[] args)
        {
            float[] a = new float[20];
            float[] b = new float[20];
            int j = a.Length - 1;

            for (int i = 0; i < a.Length; i++)
            {
                Console.WriteLine("Insira um numero na matriz A: ");
                a[i] = float.Parse(Console.ReadLine());
            }

            for (int i = 0; i < b.Length; i++)
            {
                b[i] = a[j];
                j--;
            }

            Console.Write("Matriz A: ");
            foreach (var item in a) Console.Write(item + ",");
            Console.WriteLine();
            Console.Write("Matriz B: ");
            foreach (var item in b) Console.Write(item + ",");
        }
    }
}

File: src/main/kotlin/capitulo6/ExercicioH.kt
package capitulo6

/*
 Elaborar um programa que leia 20 elementos do tipo real em uma matriz A unidimensional e construir
uma matriz B de mesma dimensão com os mesmos elementos armazenados na matriz A, porém de forma invertida.
Ou seja, o primeiro elemento da matriz A passa a ser o último da matriz B, o segundo elemento da
matriz A passa a ser o penúltimo da matriz B e assim por diante. Apresentar os elementos das matrizes
A e B.
 */
fun main() {
    val a = MutableList<Double>(20) { 0.0}
    val b = MutableList<Double>(20) { 0.0}
    var j = a.size - 1

    for (i in 0..<a.size) {
        print("Insira um número na matriz A: ")
        a[i] = readln().toDouble()
    }
    for (i in 0..<b.size) {
        b[i] = a[j]
        j -= 1
    }

    println("Matriz A:")
    for (item in a) print("$item ")
    println("\nMatriz B:")
    for (item in b) print("$item ")
}